analysers/analyser_osmosis_polygon_intersects.py

```diff
@@ -135,41 +135,67 @@
+  mobility_way_type,
+  landusage_way_id,
+  landusage_poly,
+  landusekey,
+  landusevalue,
+  tag_way,
+  tag_polygon
+FROM (
+  SELECT
+    mobility_ways.id AS mobility_way_id,
+    mobility_ways.linestring AS mobility_way_linestring,
+    mobility_ways.type AS mobility_way_type,
+    landusage.id AS landusage_way_id,
+    landusage.poly AS landusage_poly,
+    landusage.linestring AS landusage_linestring,
+    landusage.landusekey,
+    landusage.landusevalue,
+    CONCAT(mobility_ways.type, '=', mobility_ways.tags->mobility_ways.type) AS tag_way,
+    CONCAT(landusage.landusekey, '=', landusage.landusevalue) AS tag_polygon,
+    (ST_Dump(ST_Transform(ST_Intersection(landusage.poly, mobility_ways.linestring), {proj}))).geom AS intersection_part_geom,
+    landusage.is_multipolygon AS skip_intersection_size_check -- due to ST_buffer approach for mp, instead of full parsing of inners/outers
+  FROM
+    mobility_ways
+    JOIN landusage ON
       (
-        NOT landusage.tags?'layer' AND NOT mobility_ways.tags?'layer'
-      ) OR (
-        landusage.tags?'layer' AND mobility_ways.tags?'layer' AND
-        landusage.tags->'layer' = mobility_ways.tags->'layer'
-      )
-    ) AND
-    landusage.linestring && mobility_ways.linestring AND
-    -- Only find ways where the highway/railway/aeroway actually enters the
-    -- polygon. Exclude cases where the polygon is tied to the way without going inside.
-    (
-      ST_Crosses(landusage.linestring, mobility_ways.linestring) OR
+        (
+          NOT landusage.tags?'layer' AND NOT mobility_ways.tags?'layer'
+        ) OR (
+          landusage.tags?'layer' AND mobility_ways.tags?'layer' AND
+          landusage.tags->'layer' = mobility_ways.tags->'layer'
+        )
+      ) AND
+      landusage.linestring && mobility_ways.linestring AND
+      -- Only find ways where the highway/railway/aeroway actually enters the
+      -- polygon. Exclude cases where the polygon is tied to the way without going inside.
       (
-        ST_Contains(landusage.poly, mobility_ways.linestring) AND
-        NOT landusage.is_multipolygon -- due to ST_buffer approach for mp, instead of full parsing of inners/outers
+        ST_Crosses(landusage.linestring, mobility_ways.linestring) OR
+        (
+          ST_Contains(landusage.poly, mobility_ways.linestring) AND
+          NOT landusage.is_multipolygon -- due to ST_buffer approach for mp, instead of full parsing of inners/outers
+        )
       )
+  ) AS t
+WHERE
+  skip_intersection_size_check OR
+  (
+    -- Only include significant intersections
+    ST_Length(intersection_part_geom) > 2 AND
+    (
+      ST_Distance(ST_Centroid(intersection_part_geom), ST_Transform(mobility_way_linestring, {proj})) > 1 OR
+      ST_Distance(ST_Centroid(intersection_part_geom), ST_Transform(landusage_linestring, {proj})) > 1
     )
+  )
 """
 
 sql14 = """
 -- Intersections with railway, aeroway or highway
 SELECT
+  DISTINCT ON (mobility_way_id, landusage_way_id)
```

Comment:
Usually is `DISTINCT ON` + `ORDER BY`, or `GROUP BY`. Not a mix of `DISTINCT ON` + `GROUP BY`.

Reply:
Good point. I'll change it to `GROUP BY`.
(I can also use a regular `DISTINCT` and no `ORDER BY`, but this would give a double output for linestrings used in two multipolygon relations with different tags as outer. (Which, admittedly, is rare))